### PR TITLE
AP_RCProtocol: Fix the issue of ST24 receiver not working

### PR DESCRIPTION
**Pull request by lipper-chengdu**
Fix the issue of ST24 receiver not working

### CODE REVIEW

**Comment by peterbarker** [comment]
> Fix the issue of ST24 receiver not working

Wow.  On the face of it you're fixing an absolutely terrible bug.  Did you have a look to see how the bug got there?!

**Review by peterbarker — CHANGES_REQUESTED** [review]
Commit will need a message of: `AP_RCProtocol: Fix the issue of ST24 receiver not working"

Let me know if you're going to have trouble changing the commit message.

**Comment by peterbarker** [comment]
@lipper-chengdu suggest you not commit to your master branch.  See https://ardupilot.org/dev/docs/git-branch.html#creating-a-branch - always use a "topic" branch.

**Comment by lipper-chengdu** [comment]
> @lipper-chengdu suggest you not commit to your master branch. See https://ardupilot.org/dev/docs/git-branch.html#creating-a-branch - always use a "topic" branch.

I have modified the PR name. Do I need to submit the PR again using a branch?

File: libraries/AP_RCProtocol/AP_RCProtocol_ST24.cpp [hunk 1]
```diff
@@ -173,6 +173,7 @@ void AP_RCProtocol_ST24::_process_byte(uint8_t byte)
                     values[chan_index] = (uint16_t)(values[chan_index] * ST24_SCALE_FACTOR + .5f) + ST24_SCALE_OFFSET;
                     chan_index++;
                 }
+                add_input(num_values, values, false);
```
Comment (lipper-chengdu) [inline libraries/AP_RCProtocol/AP_RCProtocol_ST24.cpp:176]: AP_RCProtocol: Fix the issue of ST24 receiver not working

**Comment by peterbarker** [comment]
> > @lipper-chengdu suggest you not commit to your master branch. See https://ardupilot.org/dev/docs/git-branch.html#creating-a-branch - always use a "topic" branch.
> 
> I have modified the PR name. Do I need to submit the PR again using a branch?

The PR name is not the problem - it's the commit:
image

You do not need to re-submit using a branch - but the problem is that it's tough for you to use your git tree for anything else until this one gets merged!

**Comment by lipper-chengdu** [comment]
> > > @lipper-chengdu suggest you not commit to your master branch. See https://ardupilot.org/dev/docs/git-branch.html#creating-a-branch - always use a "topic" branch.
> > 
> > 
> > I have modified the PR name. Do I need to submit the PR again using a branch?
> 
> The PR name is not the problem - it's the commit: image
> 
> You do not need to re-submit using a branch - but the problem is that it's tough for you to use your git tree for anything else until this one gets merged!

Oh, I understand. Thank you very much

**Comment by peterbarker** [comment]
@lipper-chengdu I've taken the liberty of folding the commits and adjusting the commit message here.

This is now well-formed for merging into ArduPilot.

**Comment by lipper-chengdu** [comment]
I am not familiar with using the tool and will try resubmitting it.

> @lipper-chengdu I've taken the liberty of folding the commits and adjusting the commit message here.
> 
> This is now well-formed for merging into ArduPilot.

Thank you, by the way https://github.com/ArduPilot/ardupilot/tree/master/Tools/IO_Firmware Will the files in the directory be automatically updated? Because the modification of AP_RCProtocol_ST24.cpp file needs to regenerate these files.

**Comment by tridge** [comment]
@lipper-chengdu it would be great to get ST24 tested in our test harness:
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_RCProtocol/examples/RCProtocolTest/RCProtocolTest.cpp#L343
we need some captured serial bytes for the test harness. Can you get those?

**Review by peterbarker — APPROVED** [review]

**Comment by lipper-chengdu** [comment]
> @lipper-chengdu it would be great to get ST24 tested in our test harness: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_RCProtocol/examples/RCProtocolTest/RCProtocolTest.cpp#L343 we need some captured serial bytes for the test harness. Can you get those?
RC->FC
55 55 18 00 00 00 C4 00 00 08 00 80 08 00 80 08 00 80 08 00 80 08 00 2A 62 A6 7D //SR12
55 55 18 00 00 00 C1 00 00 08 00 80 08 00 80 08 00 80 08 00 80 08 00 2A 62 A6 E1 //SR12
55 55 2B 03 00 00 C1 00 00 08 00 80 08 00 80 08 00 80 08 00 80 08 00 2A 62 A6 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 5F//GPS.NO USE
FC->RC
55550804000042494E44B05A//BIND
55552602757800000000000000000000000000000000FFFF104000EEFFB7FFDE91FFFF110000000006//TelemetryData

I am trying to support the data retrieval function of ST24

**Comment by lipper-chengdu** [comment]
@peterbarker

Did I click the wrong button again? There's an extra committee

**Comment by peterbarker** [comment]
> @peterbarker
> 
> Did I click the wrong button again? There's an extra committee

Yep - need to rebase on master, not add a merge commit for it.  I've hit the rebase button again now so we can merge this one CI has passed.

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**